src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -373,4 +373,32 @@ trait Erasure {
       specialErasure(sym)(tp)
     }
   }
+
+  /** Does this symbol compile to the underlying platform's notion of an interface,
+    * without requiring compiler magic before it can be instantiated?
+    *
+    * More specifically, we're interested in whether LambdaMetaFactory can instantiate this type,
+    * assuming it has a single abstract method. In other words, if we were to mix this
+    * trait into a class, it should not result in any compiler-generated members having to be
+    * implemented in ("mixed in to") this class (except for the SAM).
+    *
+    * Thus, the type must erase to a java interface, either by virtue of being defined as one,
+    * or by being a trait that:
+    *   - is static (explicitouter or lambdalift may add disqualifying members)
+    *   - extends only other traits that compile to pure interfaces (except for Any)
+    *   - has no val/var members
+    */
```

Comment:
(and https://github.com/lampepfl/dotty/blob/d875fef31d984198489a3f94bdb4894698ad0cdb/src/dotty/tools/dotc/config/JavaPlatform.scala#L42-L48 and https://github.com/lampepfl/dotty/blob/d875fef31d984198489a3f94bdb4894698ad0cdb/src/dotty/tools/dotc/config/SJSPlatform.scala#L14-L16)